### ISSUE TITLE
noisy warning when a test fails in jest (and other test runners)

### ISSUE THREAD

**Issue by s101d1** [dotenvx/dotenvx#156]
I have `dotenvx` (v0.28.0) and `jest` installed in a barebone npm project, and I created an `index.test.js` file that looks like this:

```
test("testing", function () {
  expect(true).toBe(false);  // trigger a test failure
});
```
and I have empty `.env` file created in the same folder.

If I run command `npx dotenvx run -- jest`, the console would show error message like this:

> ...
> Test Suites: 1 failed, 1 total
> Tests:       1 failed, 1 total
> Snapshots:   0 total
> Time:        0.238 s, estimated 1 s
> Ran all test suites.
> 
> Command failed with exit code 1: /home/testuser/projects/nodejs/my-app/node_modules/.bin/jest
> command [jest] failed
> 
>   try without dotenvx: [jest]
> 
> if that succeeds, then dotenvx is the culprit. report issue:
> <https://github.com/dotenvx/dotenvx/issues/new>

If I run `npx jest` (not using the dotenvx), I won't get "command failed" error message:

> ...
> Test Suites: 1 failed, 1 total
> Tests:       1 failed, 1 total
> Snapshots:   0 total
> Time:        0.238 s, estimated 1 s
> Ran all test suites.

Looks like the jest command with a failed test returns non-zero status, so `dotenvx run` will interpret it as error and thus will show that error message. I think the error message `Command failed with exit code 1` might look misleading to users, it doesn't really need to be shown by default (maybe could add a verbose or debug option to enable showing that message).

Or perhaps just shows this error message:

> dotenvx run -- jest command failed with exit code 1: /home/testuser/projects/nodejs/my-app/node_modules/.bin/jest

 
The message part like `command [jest] failed ... try without dotenvx: [jest] ... if that succeeds, then dotenvx is the culprit. report issue: ...` in my opinion doesn't need to be shown, because it might sound misleading in unit test usage scenario.

**Comment by motdotla** [comment]
> If the throw new Error("hello") line is removed, no error at all for the dotenvx run.

yeah, that's because you are throwing an error in your test. you need to capture that error with something like a try/catch in your test. this is not a dotenvx issue. this is an issue with the way you've written your test.

test("testing", function () {
  try {
    throw new Error("hello");
  catch (error) {
    expect(error.message).toEql('hello')
  }
});

**Comment by s101d1** [comment]
In real life project, service/utility functions or 3rd party libraries will throw error/exception under the hood, there is no way I would wrap all of my tests with try-catch just for sake of surpressing the dotenvx's `Command failed with exit code 1` error message. Exception/Error that is thrown is a valid signal of a failed test, there is no really need to catch every error/exception in a test (unless it is a negative test). So the issue isn't about the test code itself, but the misleading `Command failed with exit code 1` error message due to Error/Exception being thrown. This issue may makes the `dotenvx run` command unfavourable to use.

**Comment by s101d1** [comment]
A little update and a correction too perhaps, actually the `Command failed with exit code 1` error message is triggered not just from a thrown Error/Exception, but a simple test failure like `expect(true).toBe(false)` will also cause the same error message. I will update the original post.

**Comment by motdotla** [comment]
@s101d1 no i don't think that is right. i've not seen that in any usage. it only occurs if a throw happens - which makes sense. dotenvx should respect any throws of errors your code has.

**Comment by s101d1** [comment]
Check my updated original post, actually the problem is not specific to Error/Exception throw at all, it turns out any failed test event (regardless how the test failure gets triggered, and jest will return non-zero status for any test failures) will trigger that `Command failed with exit code 1` error message.

**Comment by motdotla** [comment]
ah, I see your point. yes, you're correct. any process that throws a non 0 exit code will surface an error - which will subsequently surface the dotenvx warning/error. You can just ignore the dotenvx one and see the real error it surfaces. (maybe we'll think on ways to make this less visible. currently it seems like it is dotenvx's fault when it is not. but our intention there was to make sure people could debug dotenvx issues)

**Comment by nickredmark** [comment]
I just introduced dotenvx with a number of scripts that are supposed to fail from time to time (tests, build etc.) and encountering the same issue. The message being red and quite verbose makes it a small stumbling block every time to find the real issue in the output. I would be grateful if there was a way to suppress the warning. :pray:

**Comment by motdotla** [comment]
https://github.com/dotenvx/dotenvx/pull/173

available in `v0.32.0` shortly

**Comment by nickredmark** [comment]
thanks :+1:
